ecr-login/ecr.go

```diff
@@ -70,14 +71,37 @@ func NewECRHelper(opts ...Option) *ECRHelper {
 // ensure ECRHelper adheres to the credentials.Helper interface
 var _ credentials.Helper = (*ECRHelper)(nil)
 
-func (ECRHelper) Add(creds *credentials.Credentials) error {
-	// This does not seem to get called
-	return notImplemented
+func shouldIgnoreCredsStorage() bool {
+	return os.Getenv("AWS_ECR_IGNORE_CREDS_STORAGE") == "true"
 }
 
-func (ECRHelper) Delete(serverURL string) error {
-	// This does not seem to get called
-	return notImplemented
+// Add tries to store credentials when docker requests it. This usually happens during `docker login` calls. In our context,
+// storing arbitrary user given credentials makes no sense.
+func (self ECRHelper) Add(creds *credentials.Credentials) error {
+	if shouldIgnoreCredsStorage() {
+		self.logger.
+			WithField("username", creds.Username).
+			WithField("serverURL", creds.ServerURL).
+			Warning("Ignoring request to store credentials. " +
```

Comment:
Prefer to include the env. Perhaps something better:
```
Ignoring request to store credentials since AWS_ECR_IGNORE_CREDS_STORAGE env variable is set.
```

Reply:
I have retained the concatenated `This is not supported in the context of the docker ecr-login helper.` since it reflects the current situation still, though if you believe we don't need to include this, let me know!